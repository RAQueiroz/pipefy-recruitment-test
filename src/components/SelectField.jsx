import React from "react";
import styled from "styled-components";
import StyledLabel from "./StyledLabel";
import {InlineError} from "./StyledElements";
import Colors from "../utils/Colors";

const StyledSelect = styled.select`
 border: solid 1px ${props => (props.error ? Colors.errorLight : "#ccc")};
  line-height: 40px;
  height: 40px;
  border-radius: 5px;
  background-color: #fff;
  padding-left: 10px;
`;

const SelectField = ({
  label,
  id,
  error,
  options,
  value = "",
  onChange = () => {}
}) => (
  <div>
    <StyledLabel>
      {label}
      <StyledSelect
        id={id}
        error={error}
        value={value}
        onChange={event => onChange({ id, value: event.target.value })}
      >
        <option value="">Select one</option>
        {options.map(item => (
          <option value={item} key={item}>
            {item}
          </option>
        ))}
      </StyledSelect>

    </StyledLabel>
      {error && <InlineError>{error}</InlineError>}
  </div>
);

export default SelectField;
