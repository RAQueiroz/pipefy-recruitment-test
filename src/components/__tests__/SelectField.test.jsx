import React from "react";
import SelectField from "../SelectField";
import { render, Simulate, wait } from "react-testing-library";

describe(" <SelectField/>", () => {
  const item = {
    id: "xpto",
    label: "Name",
    value: "",
    options: ["A", "B", "C"]
  };

  const onChangeSpy = jest.fn();
  const { getByTestId, container } = render(
    <SelectField onChange={onChangeSpy} {...item} />
  );

  it("renders", () => {
    expect(container.querySelector("select")).toBeTruthy();
  });
  //
  it("to match snapshot", () => {
    expect(container.firstChild).toMatchSnapshot();
  });
  //

  it("render correct number of radios", () => {
    // 4 - (3 items and 1 empty one)
    expect(container.querySelector("select>option:nth-child(4)")).toBeTruthy();
  });

    it("on change to be called", () => {
        Simulate.change(container.querySelector('select'));
        expect(onChangeSpy).toHaveBeenCalled();
    });
});
