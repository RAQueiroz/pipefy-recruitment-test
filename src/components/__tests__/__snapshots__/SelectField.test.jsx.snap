// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` <SelectField/> to match snapshot 1`] = `
<div>
  <label
    class="sc-bdVaJa hOggSJ"
  >
    Name
    <select
      class="sc-gzVnrw dpetWd"
      id="xpto"
    >
      <option
        value=""
      >
        Select one
      </option>
      <option
        value="A"
      >
        A
      </option>
      <option
        value="B"
      >
        B
      </option>
      <option
        value="C"
      >
        C
      </option>
    </select>
  </label>
</div>
`;
